core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -432,8 +432,10 @@ private[spark] class ExecutorAllocationManager(
     if (testing || executorsRemoved.nonEmpty) {
       executorsRemoved.foreach { removedExecutorId =>
         newExecutorTotal -= 1
+        val hasCachedBlocks = SparkEnv.get.blockManager.master.hasCachedBlocks(executorId);
```

Comment:
Btw, there could be a chance when querying executor from BlockManager, the executor/block manager was already removed, so we will potentially get `false`.